payment/src/main/java/org/killbill/billing/payment/dispatcher/CallableWithRequestData.java

```diff
@@ -72,7 +72,8 @@ public T call() throws Exception {
             UUIDs.setRandom(null);
```

Comment:
Unrelated to your change but this whole `finally` block is a crazy amount of thread reset state!